src/tree/assess/TreeAssessor.js

```diff
@@ -0,0 +1,188 @@
+	 * @param {module:tree/assess.ScoreAggregator}  options.scoreAggregator Aggregates the scores on the individual assessments into one.
+	 * @param {module:tree/assess.Assessment[]}     [options.assessments]   The list of assessments to apply.
+	 */
+	constructor( options ) {
+		/**
+		 * A Jed object to use for translations.
+		 * @type {Jed}
+		 */
+		this.i18n = options.i18n;
+		/**
+		 * Supplies the assessments with researches and their (cached) results.
+		 * @type {module:tree/research.TreeResearcher}
+		 */
+		this.researcher = options.researcher;
+		/**
+		 * Aggregates the scores on the individual assessments into one overall score.
+		 * @type {module:tree/assess.ScoreAggregator}
+		 */
+		this.scoreAggregator = options.scoreAggregator;
+		/**
+		 * The list of assessments to apply.
+		 * @type {module:tree/assess.Assessment[]}
+		 */
+		this._assessments = options.assessments || [];
+		// Make sure that all of the assessments have the researcher.
+		this._assessments.forEach( assessment => assessment.setResearcher( this.researcher ) );
+	}
+
+	/**
+	 * Returns the list of available assessments.
+	 *
+	 * @returns {module:tree/assess.Assessment[]} The list of all available assessments.
+	 */
+	getAssessments() {
+		return this._assessments;
+	}
+
+	/**
+	 * Assesses the given text by applying all the assessments to it
+	 * and aggregating the resulting scores.
+	 *
+	 * @param {Paper}                      paper The paper to assess. This contains metadata about the text.
+	 * @param {module:tree/structure.Node} node  The root node of the tree to check.
+	 *
+	 * @returns {Promise<{results: AssessmentResult[], score: number}>} The assessment results and the overall score.
+	 */
+	async assess( paper, node ) {
+		const applicableAssessments = await this.getApplicableAssessments( paper, node );
+		/*
+		  Apply every applicable assessment on the document.
+		  Wait before they are done before aggregating the results
+		  and returning the results and final score.
+		 */
+		const results = await Promise.all(
+			applicableAssessments.map( assessment => this.applyAssessment( assessment, paper, node ) )
+		);
+		// Filter out errored assessments.
+		const validResults = results.filter( result => result.getScore() !== -1 );
+		// Compute overall score.
+		const score = this.scoreAggregator.aggregate( validResults );
+
+		return { results, score };
+	}
+
+	/**
+	 * Applies the given assessment to the paper-node combination.
+	 *
+	 * @param {module:tree/assess.Assessment} assessment The assessment to apply.
+	 * @param {Paper}                         paper      The paper to apply the assessment to.
+	 * @param {module:tree/structure.Node}    node       The root node of the tree to apply the assessment to.
+	 *
+	 * @returns {Promise<AssessmentResult>} The result of the assessment.
+	 */
+	async applyAssessment( assessment, paper, node ) {
+		return assessment.apply( paper, node ).catch(
```

Comment:
Small tidbit: When using `async` & `await` you can use `try`&`catch` if you want instead of `.catch`.